bin/run

```diff
@@ -81,14 +81,14 @@ with gfile.FastGFile(model_file, 'rb') as f:
 
 inputs = []
 for i, spec in enumerate(input_spec):
-    def scope(i: int) -> Callable[[], tf.Tensor]:
+    def scope(i: int, spec: dict) -> Callable[[], tf.Tensor]:
```

Comment:
I don't see why you need to pass `spec` here; did you run into trouble without it?

Reply:
likewise, you shouldn't need to pass `i`

Reply:
It's not exactly clear why I was running into trouble without `spec` in scope but it solved my problem. 

Let's say I have two placeholders `input_char` and `eff_history`:
`input_spec = [{'name': 'input_chars', 'dtype': 1, 'shape': [1, 400, 100]}, {'name': 'eff_history', 'dtype': 3, 'shape': [1]}]`

When I was doing `print(spec)`, after line 83, I was getting correctly the two placeholder `{'name': 'input_chars', 'dtype': 1, 'shape': [1, 400, 100]}` and `{'name': 'eff_history', 'dtype': 3, 'shape': [1]}`.

However when I was doing `print(spec)` after line 85, within `provide_input` I was getting two times {'name': 'eff_history', 'dtype': 3, 'shape': [1]} instead of getting `input_char` and `eff_history`.

When I pass `spec` to `scope` everything works properly.

Reply:
you're right, the below prints `4` six times! ([python is silly](https://eev.ee/blog/2011/04/24/gotcha-python-scoping-closures/))

good catch, forget what I said

```python
functions = []

for i in range(5):
  def function():
    return i
  functions.append(function)

print(i)

for f in functions:
  print(f())
```